### PR TITLE
Manager DNA box barcode for SMRT v12 Sequel IIe

### LINKED ISSUE #1249: DPL-814 SMRTLink v12 UI support for Sequel IIe
**User story**
As a user (Craig C) I would like to remove 'DNA Control Complex Box Barcode’ from the PacBio run setup if it is a SMRTLink v12 run as the lab no longer have a barcode to scan.

**Who are the primary contacts for this story**
Ben T, Craig C

**Who is the nominated tester for UAT**
Craig C

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] Hide ‘DNA Control Complex Box Barcode’ from the V12 PacBio run setup as the lab no longer have a barcode to scan.
- [x] Show 'DNA Control Complex Box Barcode' for old runs and V11 PacBio runs
- [x] Show well defaults correctly for v12_sequel_iie smrtlink. (may need to confirm these but should be same as v11)

**Additional context**
There is an element of service work in here as the DNA control complex box barcode is validated in the service. My advice would be to add a filter to the existing validation so that it only validates the DNA control complex barcode when it is a v11 run.

Dependent on https://github.com/sanger/traction-service/issues/1051.

### PR DESCRIPTION
**Pull request by harrietc52**
Closes #1249

Dependent on https://github.com/sanger/traction-service/issues/1051. 

Changes proposed in this pull request:

-  Hide ‘DNA Control Complex Box Barcode’ from the V12 PacBio run setup 
- Show 'DNA Control Complex Box Barcode' for old runs and V11 PacBio runs
- Show well defaults for v12_sequel_iie smrtlink (same as v11)
- Fix the setup of SMRTLinkVersions in tests
- No service changes were needed as the service PR [here](https://github.com/sanger/traction-service/pull/1077) makes the dna_control_box_barcode an optional field

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: package-lock.json
Comment (BenTopping) [inline package-lock.json:1]: I suspect this file was generated when you had node issues? Since there aren't any dependency changes in this branch I don't think this should have any changes.

File: src/components/pacbio/PacbioRunInfoEdit.vue [hunk 1]
```diff
@@ -16,6 +16,7 @@
       </traction-field-group>
 
       <traction-field-group
+        v-if="!smrtLinkVersionv12SequelIIe"
```
Comment (BenTopping) [inline src/components/pacbio/PacbioRunInfoEdit.vue:19]: I believe both v12_revio and v12_sequel_iie don't need `DNA Control Complex Box Barcode` so this could be changed to check both.

File: src/store/traction/pacbio/runCreate/run.js [hunk 2]
```diff
@@ -183,6 +183,23 @@ const hasPlateAttributes = ({ sequencing_kit_box_barcode, wells_attributes }) =>
...
 }
 
+/**
+ * @param {run} - A run object
+ * @param {smrtLinkVersion} - The SMRT Link Version of the run
+ * @returns {run} - A run object
+ */
+const SMRT_LINK_VERSION_V12_SEQUEL_IIE = 'v12_sequel_iie'
+
+const createRunPayload = (run, smrtLinkVersion) => {
```
Comment (BenTopping) [inline src/store/traction/pacbio/runCreate/run.js:193]: Is this to catch edge cases where a v12 smrtlink run has a dna_control_complex_box_barcode value by accident? Just checking because runs are created with a null value for this by default.

File: tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js [hunk 3]
```diff
@@ -75,14 +80,32 @@ describe('PacbioRunInfoEdit', () => {
     )
   })
 
-  describe('smrt link versions', () => {
-    it('returns the correct versions', () => {
-      expect(runInfo.smrtLinkVersionList.length).toEqual(2)
+  describe('#computed', () => {
+    describe('#smrtLinkVersionSelectOptions', () => {
+      it('returns the correct versions', () => {
```
Comment (BenTopping) [inline tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js:85]: Good spot on the tests.

File: tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js [hunk 4]
```diff
@@ -75,14 +80,32 @@ describe('PacbioRunInfoEdit', () => {
...
+
+        expect(runInfo.smrtLinkVersionSelectOptions).toEqual(options)
+      })
     })
 
-    it('returns smrt link version select options', () => {
-      const options = runInfo.smrtLinkVersionList.map(({ id, name }) => ({ value: id, text: name }))
-      expect(runInfo.smrtLinkVersionSelectOptions).toEqual(options)
+    describe('#smrtLinkVersionv12SequelIIe', () => {
+      it('returns false the SMRT Link version is not v12 Sequell IIewhen ', () => {
```
Comment (BenTopping) [inline tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js:100]: Sequel* and space needed between IIe and when

File: tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js [hunk 5]
```diff
@@ -91,20 +114,27 @@ describe('PacbioRunInfoEdit', () => {
       expect(store.state.traction.pacbio.runCreate.run.name).toEqual('TRACTION-RUN-3')
     })
 
-    it('dna control complex box barcode', async () => {
+    it('shows dna_control_complex_box_barcode when SMRT Link version is not v12 Sequell IIe', async () => {
```
Comment (BenTopping) [inline tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js:117]: Sequel

File: tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js [hunk 6]
```diff
@@ -91,20 +114,27 @@ describe('PacbioRunInfoEdit', () => {
...
       const input = wrapper.find('[data-attribute=dna_control_complex_box_barcode]')
       await input.setValue('DCCB1')
       expect(store.state.traction.pacbio.runCreate.run.dna_control_complex_box_barcode).toEqual(
         'DCCB1',
       )
+      expect(wrapper.text()).toContain('DNA Control Complex Box Barcode')
+    })
+    it('does not show dna_control_complex_box_barcode when SMRT Link version is v12 Sequell IIe', async () => {
```
Comment (BenTopping) [inline tests/unit/components/pacbio/PacbioRunInfoEdit.spec.js:125]: Sequel

**Review by BenTopping — CHANGES_REQUESTED** [review]
Looking good, couple minor comments. I think DNA control complex barcode is not needed for v12_revio either. Thats my fault as I should have made it more evident when I wrote the story. This change should only require minor tweaks.

File: src/store/traction/pacbio/runCreate/run.js (hunk 2 above)
Comment (harrietc52) [reply src/store/traction/pacbio/runCreate/run.js:193]: @BenTopping Yes I just thought if they select v11, add a DNA Box Barcode, then change it to v12. The DNA box barcode would still be in the runCreate store object, and therefore sent to the Service. As the DNA Box Barcode is optional in the Service, this would let the v12 run be created with it. So, I thought to explicitly remove it here to make sure it isn't persisted.

Comment (BenTopping) [reply src/store/traction/pacbio/runCreate/run.js:193]: Yeah thats fair enough. We do run an action called 'setSmrtLinkVersion' when the smrtlink version is changed. We could add clear the dcbb there depending on the smrtLinkVersion? I dont know if thats a better solution or not though.

Comment (harrietc52) [reply src/store/traction/pacbio/runCreate/run.js:194]: @BenTopping Ye sure I wondered the same. The mutation that the `setSmrtLinkVersion` calls is very specific to just replacing the stores `smrtLinkVersion`. So in the end I thought the mutation is better of just having that one task to keep it clean, rather than the mutation changing two different objects in the store.

File: .release-version [hunk 7]
```diff
@@ -1 +1 @@
-8.0.1
```
Comment (BenTopping) [inline .release-version:1]: Only need to increase release-version on develop -> master for traction-ui

File: tests/unit/store/traction/pacbio/runCreate/run.spec.js [hunk 8]
```diff
@@ -246,7 +251,7 @@ describe('run.js', () => {
 
   describe('createPayload', () => {
     it('will create a new run payload', () => {
-      const run = { system_name: 'Revio' }
+      const run = { system_name: 'Revio', dna_control_complex_box_barcode: 'to keep' }
```
Comment (BenTopping) [inline tests/unit/store/traction/pacbio/runCreate/run.spec.js:254]: I would change this system name to Sequel IIe as that is what is being selected as instrument type on line 261.

File: tests/unit/store/traction/pacbio/runCreate/run.spec.js [hunk 9]
```diff
@@ -337,5 +344,71 @@ describe('run.js', () => {
...
+
+    it('will create the correct payload when SMRT Link Version is v12 Revio', () => {
+      const run = { system_name: 'Revio', dna_control_complex_box_barcode: 'redundant' }
+
+      const payload = createPayload({
+        run,
+        plates: plates.new,
+        wells: wells.new,
+        smrtLinkVersion: smrtLinkVersions['2'],
+        instrumentType: PacbioInstrumentTypes.SequelIIe,
```
Comment (BenTopping) [inline tests/unit/store/traction/pacbio/runCreate/run.spec.js:389]: Instrument type should be Revio

**Review by BenTopping — CHANGES_REQUESTED** [review]

**Review by BenTopping — APPROVED** [review]
